src/main/java/banking/BankingSystem.java

```diff
@@ -0,0 +1,200 @@
+
+            processInput(option);
+            syncAccount();
+        } while (!"0".equals(option));
+    }
+
+    private void showMenu() {
+        if (!loggedIn) {
+            System.out.printf("""
+                    %n1. Create an account
+                    2. Log into account
+                    """);
+        } else {
+            System.out.printf("""
+                    %n1. Balance
+                    2. Add income
+                    3. Do transfer
+                    4. Close account
+                    5. Log out
+                    """);
+        }
+        System.out.println("0. Exit\n");
+    }
+
+    private void processInput(String input) {
+        if (loggedIn) {
+            processLoggedOption(input);
+        } else {
+            processBasicOption(input);
+        }
+    }
+
+    private void syncAccount() {
+        if (currentAccount != null) {
+            Optional<Account> account = bank.signIn(
+                    currentAccount.card().cardNumber(),
+                    currentAccount.card().PIN());
+
+            account.ifPresentOrElse(
+                    this::setCurrentAccount,
+                    () -> {
+                        logOut();
+                        System.out.println("\nError, probably card PIN has been changed. Try to sign in again.");
+                    }
+            );
+        }
+    }
+
+    private void processLoggedOption(String option) {
+        switch (option) {
+            case "1" -> showBalanceOption();
+            case "2" -> addIncomeOption();
+            case "3" -> doTransferOption();
+            case "4" -> closeAccountOption();
+            case "5" -> logOutOption();
+            case "0" -> System.out.println("\nBye!\n");
+            default -> System.out.println("\nIncorrect option. Try again.");
+        }
+    }
+
+    private void processBasicOption(String option) {
+        switch (option) {
+            case "1" -> createAccountOption();
+            case "2" -> logInOption();
+            case "0" -> System.out.println("\nBye!\n");
+            default -> System.out.println("Incorrect option. Try again.");
+        }
+    }
+
+    private void showBalanceOption() {
+        System.out.println("Balance: " + currentAccount.balance());
+    }
+
+    private void addIncomeOption() {
+        System.out.println("\nEnter income:");
+
+        try {
+            int income = scanner.nextInt();
+            if (income < 1) {
+                throw new Exception();
+            }
+
+            bank.addIncome(currentAccount.card().cardNumber(), income);
+            System.out.println("Income was added!");
+        } catch (Exception ex) {
+            System.out.println("Incorrect input. Enter a number greater than zero!");
+        }
+    }
+
+    private void doTransferOption() {
+        System.out.println("\nTransfer");
+        System.out.println("Enter card number:");
+
+        try {
+            String receiver = scanner.next();
```

Comment:
Maybe you can move validation into some other place?

Reply:
Done

Reply:
Okay, decomposed this method into several: `canTransferTo` and `canSend` methods